search_test.go

```diff
@@ -87,3 +89,44 @@ func TestSearch_ExecDo(t *testing.T) {
 	}
 
 }
+
+func TestSearch_PartialExec(t *testing.T) {
```

Comment:
Can you add a test for what happens when 1001 records total will be returned? That would take two requests to the server, correct?

Reply:
1001 isn't a magic number.  It's just hard coded in too many places.  I rewrote the code to make that number settable.  When and if you might want to grab my version.

Reply:
@tyler-ball Would you still like this test case to be added? We are overriding the HTTP calls / response with mock JSON for unit tests. Probably the tricky part will be to figure out how to distinguish between the first and second call to override the responses correctly, but I can look at it, let me know.

Reply:
@tyler-ball added a new test case to verify multiple http calls and combined response via partial search.